### ISSUE TITLE
InfNumFastPopsExpire -> Primewire.ag

### ISSUE THREAD

**Issue by forkoz** [uBlockOrigin/uAssets#521]
Replace the bracketed [...] placeholders with your own information.

### URL(s) where the issue occurs

primewire.ag

### Describe the issue

So primewire.ag redirects/popups are back with a vengance.

Previously you were able to set InfNumFastPopsExpire to 2020 or some other time and they would go away. Now advancing the date causes the redirect immediately. Setting it back is safe while sitting on the page but it gets re-created if you click anything.

Filters are useless because a bunch of random character domains were registered and they rotate.
ie. 153105c2f9564.com/1304358.html?q

A new variable is also there: InfNumFastPops Not sure what it does.

### Versions

- Browser/version: [FF/Chromium]
- uBlock Origin version: [1.13.4]

### Settings

- [List here all the changes you made to uBO's default settings]

### Notes

Tried scripts but they did not work. 
![primewire](https://user-images.githubusercontent.com/12354546/28244917-6121d1d6-69be-11e7-96e1-7cb0b865785c.png)

**Comment by okiehsch** [comment]
I can only reproduce if I disable the `uBO filter` list, so this seems already fixed, probably by this
[commit](https://github.com/uBlockOrigin/uAssets/commit/184a33f7364f342d77a625955d7a4d66335af534)

**Comment by uBlock-user** [comment]
Try - `primewire.ag##script:inject(csp.js, script-src 'self')`

With that, it doesn't make any such connections to random domains anymore.

**Comment by gorhill** [comment]
I think this can help further, as some video links in there are meant to open those unwanted popups (which were auto-closed by uBO, but still):

    primewire.ag##.actual_tab > table:if(td:first-of-type + td + td:has-text(/Promo/))

**Comment by gorhill** [comment]
Need confirmation the issue is solved.

**Comment by forkoz** [comment]
?? Its happening on chromium and ff. That commit doesn't have any of the URLs which pop up.

I have all of these enabled

```
 My filters​​​​​43 used out of 47​​​​
uBlock filters​
​​​​1,496 used out of 1,526​​
​​
uBlock filters – Badware risks​
​​​​7 used out of 8​​
​​
uBlock filters – Experimental​
​​​​6 used out of 6​​
​​
uBlock filters – Privacy​​​​​50 used out of 59​​
​​
uBlock filters – Unbreak​​​​​172 used out of 175​​

    ​​ 

Ads (4)

    Adblock Warning Removal List​

​​​​363 used out of 383​​
​​
Anti-Adblock Killer | Reek​
​​​​61 used out of 2,179​​
​​
EasyList​
​​​​0 used out of 75,093​​
​​
EasyList without element hiding rules​
​​​​​​​​
uBlock Protector List​
​​​​2,485 used out of 2,485​​

    ​​ 

Privacy (3)

    Basic tracking list by Disconnect​​​​​0 used out of 34​​

​​
EasyPrivacy​
​​​​5 used out of 13,303​​
​​
Fanboy’s Enhanced Tracking List​
​​​​0 used out of 1,976​​

    ​​ 

Malware domains (6)

    Malvertising filter list by Disconnect​​​​​182 used out of 5,067​​

​​
Malware Domain List​​​​​1,107 used out of 1,152​​
​​
Malware domains​
​​​​24,580 used out of 26,952​​
​​
Malware domains (long-lived)​
​​​​2,605 used out of 3,204​​
​​
Malware filter list by Disconnect​​​​​2,313 used out of 2,331​​
​​
Spam404​
​​​​5,815 used out of 5,850​​

    ​​ 

Social (3)

    Anti-ThirdpartySocial (see warning inside list)​

​​​​47 used out of 66​​
​​
Fanboy’s Annoyance List​
​​​​1 used out of 26,237​​
​​
Fanboy’s Social Blocking List​
​​​​0 used out of 15,397​​

    ​​ 

Multipurpose (5)

    Dan Pollock’s hosts file​

​​​​9,123 used out of 12,847​​
​​
Fanboy+Easylist-Merged Ultimate List​
​​​​114,776 used out of 116,603​​
​​
hpHosts’ Ad and tracking servers​
​​​​43,898 used out of 48,013​​
​​
MVPS HOSTS​
​​​​12,597 used out of 13,090​​
​​
Peter Lowe’s Ad and tracking server list​
​​​​2,327 used out of 2,327​​​​
```

**Comment by forkoz** [comment]
> Try - primewire.ag##script:inject(csp.js, script-src 'self')

That works but it kills the sorting and genre menu. I can do the same thing just turning off inline scripts.

Second filter works well. Puts the primewire links .1 script out of business.

**Comment by uBlock-user** [comment]
This fixes the sorting and genre menu

`primewire.ag##script:inject(csp.js, script-src 'self' 'unsafe-inline')`

**Comment by forkoz** [comment]
I still get the popups with that but yes it preserves the sorting. I think closing is premature :)

First filter also kills the links:
![primewire2](https://user-images.githubusercontent.com/12354546/28248629-37305394-6a0d-11e7-8dfd-05413e01a8fe.png)

**Comment by gorhill** [comment]
> First filter also kills the links

It removes that nasty links, these were promos causing popup ads to open in new tabs.

**Comment by gorhill** [comment]
@uBlock-user 

> `primewire.ag##script:inject(csp.js, script-src 'self' 'unsafe-inline')`

If you're going to suggest CSP-based filters, please suggest the more resilient `csp=` one. The `script:inject` one is subject to race conditions, i.e. no guarantee it will apply before whatever it is meant to blocked is executed.

**Comment by forkoz** [comment]
No, these are the real links. Your previous filter killed the "sponsored ones". Primewire loads the CDN names and genre stuff from an external script and "uBlock-user" filters break that.

This is the popup code: https://pastebin.com/gsXFrhBT

**Comment by uBlock-user** [comment]
I don't get any popups with that filter and it doesn't make any connections to random domains.

@gorhill 

> If you're going to suggest CSP-based filters, please suggest the more resilient csp= version.

Is that the network filter directive `$csp` ? I haven't experienced any race conditions with `csp.js` as I have been using many for quite some time, but will do if you say so.

**Comment by gorhill** [comment]
> No, these are the real links.

![a](https://user-images.githubusercontent.com/585534/28248717-fa2b7fdc-6a16-11e7-961d-79eb58e2464a.png)

Please make the case something is wrong with the filters, do not post code on pastebin, we are all well equipped to look this up for ourselves if we want.

**Comment by forkoz** [comment]
Yep, your filter is good but doesn't solve the fastpops. 2 issues here.

1. Fastpops working on a timer to popup virus scammers or redirect the page you are on.
2. Promo host links which are taken care of the filter or the user script.

**Comment by gorhill** [comment]
Since it works for both me and @okiehsch, you will have to investigate on your side. As said, it works with uBO default settings + default up to date lists. If you have custom filters or other interfering extensions, you will have to fix this yourself. uBO uses redirection to solve the issue here, so if another blocker blocks what uBO wants to redirect, this will sabotage uBO's solution. Now please, if you want to make the case there is something wrong with the filters, provide more than just : "nope doesn't work", use the logger to investigate what is happening.

**Comment by forkoz** [comment]
Ok, I will try to log and see what's happening with that filter. I only have UBO on this browser, no other blockers. The popups aren't instant. They pop a few minutes in the future. So if you don't use the site and just go to it you would not notice.

Btw, your filter needs a complimient:

> primewire.ag##.actual_tab > table:if(td:first-of-type + td + td:has-text(Sponsor))

**Comment by gorhill** [comment]
Ok, I just got a case where the link is named "Sponsor Host" instead of "Promo Host", I will update the cosmetic filter accordingly.

**Comment by forkoz** [comment]
This is the log where it pops up anyway even though things are "blocked".

![primewire3](https://user-images.githubusercontent.com/12354546/28248841-dd1551e4-6a10-11e7-87f3-1db017f18e96.png)

**Comment by gorhill** [comment]
Were on the page did you click to get that popup?

**Comment by forkoz** [comment]
You can click anywhere. Episode, TV section, movie section, music section. If InfNumFastPopsExpire is expired it will popup or redirect if popups are blocked.

**Comment by uBlock-user** [comment]
That's a hyperlink based popup. Entire CSS is one hyperlink waiting for you to click to trigger and then load a random domain.

`primewire.ag##a[href^="//"]`

**Comment by forkoz** [comment]
True, I did notice that sometimes it triggers when I just click anywhere. If InfNumFastPopsExpire is past you get a popup/redirect. If not "nothing" is happening.

**Comment by uBlock-user** [comment]
The script you posted creates an invisible overlay on the CSS and when you click anywhere it pops up + redirects you to a random domain, that last filter I posted kills the overlay from loading.

**Comment by gorhill** [comment]
> That's a hyperlink based popup

How come I don't get it, however I try.

**Comment by uBlock-user** [comment]
It's cookie based ? Remove all cookies and site data and browse some movie links or random links and you should be able to reproduce it.

**Comment by gorhill** [comment]
I know about cookies, I always clear them when trying to reproduce.

I was testing in Google Chrome and decide to try with Chromium. I was able to reproduce. I then just forced an update of EasyList + uBO filters, and can't reproduce anymore.

**Comment by uBlock-user** [comment]
I'm on Chromium v61.0.3159.0 and I can reproduce it every time I remove cookies, kill the tab and open it again in a new tab.

**Comment by forkoz** [comment]
Can you see the date in InfNumFastPopsExpire, it will only trigger after that.

I purged and re-downloaded all at the start. Filter doesn't fix it. The storage is still set even with

`primewire.ag##script:inject(csp.js, script-src 'self' 'unsafe-inline')`

so that overlay still runs.

**Comment by uBlock-user** [comment]
On my end, `InfNumFastPopsExpire` is not created after I add the filter I posted.

**Comment by forkoz** [comment]
I should try it on chromium. I'm testing FF here.

`primewire.ag##script:inject(csp.js, script-src 'self')`

Makes the variable disappear but then you lose functionality.

**Comment by gorhill** [comment]
> Can you see the date in `InfNumFastPopsExpire`, it will only trigger after that.

That is uBO setting that date. There is [this filter](https://github.com/uBlockOrigin/uAssets/blob/master/filters/resources.txt#L585) in uBO's filters:

  primewire.*##script:inject(pornhub-popup-defuser.js)

It's a script to overwrite the `InfNumFastPops` and `InfNumFastPopsExpire` cookies. If I disable that filter, I can get the popups.

There a bug in there, the value was meant to be `84600000`, not `84600`. Regardless, this is what should be preventing the popups, because the `InfNumFastPops` is set every page load.

**Comment by forkoz** [comment]
That's just like the primewire enhancer script. Problems here are:

1. Advancing the date to 2018, 2020, etc now produces the popup immediately.
2. The script and I'm assuming that filter run and then ajax updates the "correct" date after the page has loaded. I had already tried to edit the "enhancer" script to subtract a year, it kept getting overwritten. 

Also, on chromium both of "uBlock-user"'s filters break genre, the second filter at least lets you see the link CDNs.

**Comment by gorhill** [comment]
Another issue I realize is wrong with the scriptlet is that the expiration date must be specified as a GTM-based time, while it uses local time. Need to fix this.

**Comment by okiehsch** [comment]
> This is the log where it pops up anyway even though things are "blocked".
![primewire3](https://user-images.githubusercontent.com/12354546/28248841-dd1551e4-6a10-11e7-87f3-1db017f18e96.png)

@forkoz this issue is fixed by `||go.oclasrv.com/apu.php$script,redirect=noopjs`, added yesterday
 to uBO-filters.
I can reproduce the popups with Chrome and Firefox only if I disable that filter.

**Comment by forkoz** [comment]
I had already globally blocked oclasrv.com

**Comment by okiehsch** [comment]
Blocking `oclasrv.com` triggers the randomly named css overlay, nooping it fixes the issue, atleast on my end.

**Comment by forkoz** [comment]
I will try unblocking it and see. So far in FF it seems to have helped. Will try on chrome.

**Comment by okiehsch** [comment]
Here is the logger if I **noop** `oclasrv.com`
![prime](https://user-images.githubusercontent.com/16838044/28249570-921ff05c-6a58-11e7-915a-1683d47866da.PNG)
notice no randomly named css.
Here is  the logger if I **block** `oclasrv.com`
![prime1](https://user-images.githubusercontent.com/16838044/28249581-bfa6005c-6a58-11e7-88f7-d37538cf5972.PNG)
notice the randomly named css which will trigger a popup if you click anywhere.

**Comment by forkoz** [comment]
Yea, here it loads before that server. Doesn't even load on the initial page.
![primewire4](https://user-images.githubusercontent.com/12354546/28249645-49d64b22-6a1f-11e7-8d22-5857c72a8f1e.png)

**Comment by okiehsch** [comment]
Well I can't reproduce with default settings or if I try emulating your logger-output, so I can't help sorry.

**Comment by forkoz** [comment]
I can reproduce on linux and windows in both chrome and FF. 
FF only has ublock, chrome has scriptsafe, ublock, umatrix and now UBO extra.

This has been plaguing me for at least a few weeks. I even took off all remaining domain based blocks and no changes. Considering bare primewire.ag never even loads oclasrv.com, I can say its unrelated.

**Comment by gorhill** [comment]
I added a new scriplet to cancel attempts to install event listeners. Seems to work on my side -- tested after disabling uBO's filters.

**Comment by gorhill** [comment]
Furthermore, I see two exception filters in EasyList for that site:

    @@||engine.*/tag.engine?$script,domain=primewire.ag|primewire.in
    @@||cdn.engine.*^guid^$third-party,script,domain=primewire.ag|primewire.in

I have no clue as to their purpose, I see nothing broken when blocking them with the uBO's firewall. I can see the domain excepted as a result of these filters being referenced in the popup/popunder inline javascript code, so I think I will just mark these filters as `badfilter`, I will guess they were added for ABP/AB.

**Comment by forkoz** [comment]
Looks like they are gone. For those filters it may be the category/genre selection and the CDNs. Those are a 3rd party script. otherwise the dropdowns don't work and links are all un-named

![primewire](https://user-images.githubusercontent.com/12354546/28249976-50ac6eb2-6a25-11e7-9181-acdfc2d5d0b9.png)

**Comment by forkoz** [comment]
links are still fine, but sorting is what those were for

**Comment by okiehsch** [comment]
The last commit will break dropdown functionality for "Section" "Genre" "Sort By"
`http://www.primewire.ag/?tv`

**Comment by forkoz** [comment]
ding dong the witch is dead. Thank you I think that did it until they take action again.
